### PR TITLE
Add claim configuration view

### PR DESCRIPTION
**Pull request by tharindu-bandara**
## Purpose
- Addresses https://github.com/wso2/identity-apps/issues/354.

## Goals
- $subject.

## Approach
1. Configure claim mapping.
2. Configure the subject and role claims.
3. Configure provisional claim configuration.
4. Use mapped claims from step 1, in steps 2 and 3 if claim mappings available. Otherwise, fall back to local claims.

attribute-view

### CODE REVIEW

File: apps/admin-portal/src/components/identity-providers/settings/attribute-management/attribute-list-item.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+interface AttributeListItemPropInterface {
+    attribute: IdentityProviderClaimInterface;
+    placeholder: string;
+    updateMapping?: (mapping: IdentityProviderCommonClaimMappingInterface) => void;
+    mapping?: string;
+}
+
+/**
+ * Selected Attribute list item.
+ * @param props AttributeListItemPropInterface
```
Comment (inthirakumaaran) [inline apps/admin-portal/src/components/identity-providers/settings/attribute-management/attribute-list-item.tsx:34]:
```suggestion

 * @param props AttributeListItemPropInterface
```

File: apps/admin-portal/src/components/identity-providers/settings/attribute-management/attribute-selection-wizard.tsx [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+    } = props;
+
+
+    const [tempAvailableClaims, setTempAvailableClaims] = useState<IdentityProviderClaimInterface[]>([]);
+    const [filterTempAvailableClaims, setFilterTempAvailableClaims] = useState<IdentityProviderClaimInterface[]>([]);
+    
+    const [tempSelectedAttributes, setTempSelectedClaims] = useState<IdentityProviderClaimInterface[]>([]);
+    const [filterTempSelectedClaims, setFilterTempSelectedClaims] = useState<IdentityProviderClaimInterface[]>([]);
+    
+    const [checkedUnassignedListItems, setCheckedUnassignedListItems] = useState<IdentityProviderClaimInterface[]>([]);
```
Comment (inthirakumaaran) [inline apps/admin-portal/src/components/identity-providers/settings/attribute-management/attribute-selection-wizard.tsx:60]: Is this length okay?

File: apps/admin-portal/src/components/identity-providers/settings/attribute-management/role-mapping-settings.tsx [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+
+    useEffect(() => {
+        getRolesList(null)
+            .then((response) => {
+                if (response.status === 200) {
+                    const allRole: RoleListInterface = response.data;
+                    setRoleList(allRole.Resources);
+                }
+            })
+            .catch(() => {
```
Comment (inthirakumaaran) [inline apps/admin-portal/src/components/identity-providers/settings/attribute-management/role-mapping-settings.tsx:95]: Enhance the catch statement. Please check other places as well.

File: apps/admin-portal/src/components/identity-providers/settings/attribute-management/uri-attributes-settings.tsx [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+    return (
+        <>
+            <Grid.Row>
+                <Grid.Column mobile={ 16 } tablet={ 16 } computer={ 10 }>
+                    <Divider/>
+                    <Divider hidden/>
+                </Grid.Column>
+                <Grid.Column mobile={ 16 } tablet={ 16 } computer={ 8 }>
+                    {/* eslint-disable-next-line @typescript-eslint/no-empty-function */}
+                    <Forms submitState={ triggerSubmit } onSubmit={ (): void => { } }>
```
Comment (inthirakumaaran) [inline apps/admin-portal/src/components/identity-providers/settings/attribute-management/uri-attributes-settings.tsx:63]: Can't we use the Form as the outermost element?

Comment (inthirakumaaran) [reply apps/admin-portal/src/components/identity-providers/settings/attribute-management/uri-attributes-settings.tsx:63]: If they are updating for each change we can get rid of the Form or it is better to have proper onSubmit function where we update those values

File: apps/admin-portal/src/components/identity-providers/utils/attribute-settings-utils.tsx [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+}
+
+export const LocalDialectURI = "http://wso2.org/claims";
+
+export const getLocalDialectURI = (): string => {
+
+    let localDialect = "http://wso2.org/claims";
+    getAllLocalClaims(null)
+        .then((response) => {
+            // setClaims(response.slice(0, 10));
```
Comment (inthirakumaaran) [inline apps/admin-portal/src/components/identity-providers/utils/attribute-settings-utils.tsx:47]:
```suggestion
          
```

File: apps/admin-portal/src/components/identity-providers/utils/attribute-settings-utils.tsx [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+                level: AlertLevels.ERROR,
+                message: "Update error"
+            }));
+        });
+
+    updateIDPRoleMappings(idpId, {
+            mappings: roleMapping,
+            outboundProvisioningRoles: [""]
+        } as IdentityProviderRolesInterface
+    ).then(() => {
```
Comment (inthirakumaaran) [inline apps/admin-portal/src/components/identity-providers/utils/attribute-settings-utils.tsx:177]:
```suggestion
    )
       .then(() => {
```

Comment (inthirakumaaran) [reply apps/admin-portal/src/components/identity-providers/utils/attribute-settings-utils.tsx:177]: Check the format

File: apps/admin-portal/src/components/identity-providers/settings/attribute-management/attribute-selection-wizard.tsx [hunk 7]
```diff
@@ -0,0 +1,319 @@
...
+    const [checkedAssignedListItems, setCheckedAssignedListItems] = useState<IdentityProviderClaimInterface[]>([]);
+    
+    const [isSelectUnassignedClaimsAllClaimsChecked, setSelectUnassignedClaimsAllClaimsChecked] = useState(false);
+    const [isSelectAssignedAllClaimsChecked, setIsSelectAssignedAllClaimsChecked] = useState(false);
+
+    const handleAttributeModal = () => {
+        setShowAddModal(false);
+    };
+
+    // search operation for available claims.
```
Comment (inthirakumaaran) [inline apps/admin-portal/src/components/identity-providers/settings/attribute-management/attribute-selection-wizard.tsx:70]:
```suggestion
    // Search operation for available claims.
```

File: apps/admin-portal/src/components/identity-providers/settings/attribute-management/attribute-selection-wizard.tsx [hunk 8]
```diff
@@ -0,0 +1,319 @@
...
+        const changeValue = event.target.value;
+        if (changeValue.length > 0) {
+            setFilterTempAvailableClaims(filterTempAvailableClaims.filter((item) =>
+                item.uri.toLowerCase().indexOf(changeValue.toLowerCase()) !== -1))
+        } else {
+            setFilterTempAvailableClaims(tempAvailableClaims);
+        }
+    };
+
+    // search operation for selected claims.
```
Comment (inthirakumaaran) [inline apps/admin-portal/src/components/identity-providers/settings/attribute-management/attribute-selection-wizard.tsx:81]:
```suggestion
    // Search operation for selected claims.
```

File: apps/admin-portal/src/components/identity-providers/settings/attribute-management/attribute-selection-wizard.tsx (hunk 2 above)
Comment (tharindu-bandara) [reply apps/admin-portal/src/components/identity-providers/settings/attribute-management/attribute-selection-wizard.tsx:60]: Yes. It stays within 120 characters.

File: apps/admin-portal/src/components/identity-providers/utils/attribute-settings-utils.tsx (hunk 6 above)
Comment (tharindu-bandara) [reply apps/admin-portal/src/components/identity-providers/utils/attribute-settings-utils.tsx:177]: This is already formatted.

File: apps/admin-portal/src/components/identity-providers/settings/attribute-management/attribute-selection-wizard.tsx (hunk 8 above)
Comment (tharindu-bandara) [reply apps/admin-portal/src/components/identity-providers/settings/attribute-management/attribute-selection-wizard.tsx:81]: Fix review comments

File: apps/admin-portal/src/components/identity-providers/settings/attribute-management/role-mapping-settings.tsx (hunk 3 above)
Comment (tharindu-bandara) [reply apps/admin-portal/src/components/identity-providers/settings/attribute-management/role-mapping-settings.tsx]: Fixed in 73f0cc02c52d71ca5acc51cf4120b806799a9d3d

File: apps/admin-portal/src/components/identity-providers/settings/attribute-management/uri-attributes-settings.tsx (hunk 4 above)
Comment (tharindu-bandara) [reply apps/admin-portal/src/components/identity-providers/settings/attribute-management/uri-attributes-settings.tsx]: Fixed in 73f0cc02c52d71ca5acc51cf4120b806799a9d3d

**Pull request merged**